### PR TITLE
fmt: add with_os_api option

### PR DESCRIPTION
**Pull request by SpaceIm**
Specify library name and version:  **lib/1.0**

closes https://github.com/conan-io/conan-center-index/issues/5771

---

- [ ] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [ ] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [ ] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [ ] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/fmt/all/conanfile.py [hunk 1]
```diff
@@ -13,8 +15,20 @@ class FmtConan(ConanFile):
...
     generators = "cmake"
     settings = "os", "compiler", "build_type", "arch"
-    options = {"shared": [True, False], "header_only": [True, False], "fPIC": [True, False], "with_fmt_alias": [True, False]}
-    default_options = {"shared": False, "header_only": False, "fPIC": True, "with_fmt_alias": False}
+    options = {
+        "header_only": [True, False],
+        "shared": [True, False],
+        "fPIC": [True, False],
+        "with_fmt_alias": [True, False],
+        "with_os": [True, False],
```
Comment (madebr) [inline recipes/fmt/all/conanfile.py:23]: I don't really like the `with_os` name. It's too close to `self.settings.os`
See https://github.com/fmtlib/fmt/issues/1654#issuecomment-621218600
How about `with_os_api` or something similar?

File: recipes/fmt/all/conanfile.py [hunk 2]
```diff
@@ -13,8 +15,20 @@ class FmtConan(ConanFile):
...
+    }
+    default_options = {
+        "header_only": False,
+        "shared": False,
+        "fPIC": True,
+        "with_fmt_alias": False,
+        "with_os": True,
```
Comment (madebr) [inline recipes/fmt/all/conanfile.py:30]: Maybe disable the option in ´config_options´ when os not in Windows, Linux, FreeBSD, or not tools.is_apple_os?

File: recipes/fmt/all/conanfile.py (hunk 1 above)
Comment (jwillikers) [reply recipes/fmt/all/conanfile.py:23]: That's a good point. I second `with_os_api`.

File: recipes/fmt/all/conanfile.py (hunk 2 above)
Comment (jwillikers) [reply recipes/fmt/all/conanfile.py:30]: I'm not sure how likely it is, but I can imagine use cases where someone may not want the OS support compiled in to the library when targeting Windows, Linux, FreeBSD, etc. One such reason is minimizing the size of the binary.

File: recipes/fmt/all/conanfile.py (hunk 1 above)
Comment (SpaceIm) [reply recipes/fmt/all/conanfile.py]: done

**Comment by madebr** [comment]
I think you need to disable parallel builds in the test package (`cmake.parallel = False`).

**Comment by SpaceIm** [comment]
Tedious, I would like to avoid to disable parallel for a CI issue

**Comment by madebr** [comment]
> 
> 
> Tedious, I would like to avoid to disable parallel for a CI issue

Understandable, that's what I did for boost.

**Review by prince-chrismc — APPROVED** [review]

**Comment by prince-chrismc** [comment]
Another candidate for removing older versions

**Review by madebr — APPROVED** [review]

**Comment by prince-chrismc** [comment]
There are versions missing from the build... do we need a rebase?

**Review by uilianries — APPROVED** [review]

**Review by jwillikers — APPROVED** [review]

**Pull request merged**